### PR TITLE
Fix the hash generation by adding the current time

### PR DESCRIPTION
**Pull request by 99999989**
As the generated hash was not taking into account that texts can contain highlight tags (`<em>`), the yellow highlighting was not working anymore. I've added the current dateTime to the hash to make it unique. 
I've basically tested the performance in IE and it seems OK.

### CODE REVIEW

**Comment by pavax** [comment]
We need to check if this Bugfix has a Performance Impact on Internet Exporer (you can check by going to the job-search and load 100 jobs)

File: alv-portal-ui/src/app/job-advertisement/shared/job-search-result/job-search-result.component.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ function hashCode(jobSearchResult: JobSearchResult) {
     id += jobSearchResult.favouriteItem.id;
     id += jobSearchResult.favouriteItem.note;
   }
+  id += Date.now().toString();
```
Comment (birom) [inline alv-portal-ui/src/app/job-advertisement/shared/job-search-result/job-search-result.component.ts]: If we concatenate a random string (time stamp), we do not need any other info such as ``visited``, ``favouriteItem.id``, ``favouriteItem.note`` etc.. I would keep only ``jobSearchResult.jobAdvertisement.id``+``Date.now().toString()``

File: alv-portal-ui/package.json [hunk 2]
```diff
@@ -74,6 +74,7 @@
     "open-sans-fonts": "1.5.1",
     "pretty-bytes": "5.1.0",
     "rxjs": "6.4.0",
+    "xxhashjs": "^0.2.2",
```
Comment (pavax) [inline alv-portal-ui/package.json]: fix the version

File: alv-portal-ui/src/app/job-advertisement/shared/job-search-result/job-search-result.component.ts [hunk 3]
```diff
@@ -30,7 +31,8 @@ function hashCode(jobSearchResult: JobSearchResult) {
     id += jobSearchResult.favouriteItem.id;
     id += jobSearchResult.favouriteItem.note;
   }
-  return id;
+  id += JSON.stringify(jobSearchResult.jobAdvertisement.jobContent.jobDescriptions);
```
Comment (pavax) [inline alv-portal-ui/src/app/job-advertisement/shared/job-search-result/job-search-result.component.ts]: we could also try to stringify the complete jobSearchResult

**Comment by pavax** [comment]
tested in IE11 and with the hash function it is still fast enough

**Pull request merged**